### PR TITLE
MWPW-1351731 support linked text

### PR DESCRIPTION
**Pull request by VKniaz**
Title: Cta as linked text

Authors can link text strings with checkout links.
 
Screenshot 2024-01-17 at 12 48 37 PM

Resolves: [MWPW-135731](https://jira.corp.adobe.com/browse/MWPW-135731)

**Test URLs:**
- Before: https://main--milo--adobecom.hlx.page/drafts/vkniaziev/document9?martech=off
- After: https://mwpw-135731--milo--vkniaz.hlx.page/drafts/vkniaziev/document9?martech=off

### CODE REVIEW

File: libs/blocks/merch/merch.js [hunk 1]
```diff
@@ -122,11 +122,14 @@ export async function buildCta(el, params) {
   const context = await getCheckoutContext(el, params);
   if (!context) return null;
   const service = await initService();
+  const isCta = el.textContent?.startsWith('CTA');
```
Comment (Axelcureno) [inline libs/blocks/merch/merch.js:125]: Why not reuse from line 144 `const isCta = searchParams.get('type') === 'checkoutUrl';`

Comment (VKniaz) [reply libs/blocks/merch/merch.js:125]: Because in both cases, we will have the same type as the checkout URL.

**Review by Axelcureno — APPROVED** [review]

**Review by JasonHowellSlavin — APPROVED** [review]
Looks good. Nit: might be nice to have a test testing the other outcome of the `if`. Testing when the text does not start with CTA? 

```
// Mock
 <p>
    No cta here: <a class="merch linked text incorrect"
    href="/tools/ost?osi=16&type=checkoutUrl">Buy now and save 40%</a>
  </p>
```

**Comment by VKniaz** [comment]
Hey, @JasonHowellSlavin. There are tests for both cases. So, I added a case when it does not start with CTA, and there is an existing one that verifies when it does.

**Review by yesil — APPROVED** [review]
minor comments.

File: libs/blocks/merch/merch.js [hunk 2]
```diff
@@ -122,11 +122,14 @@ export async function buildCta(el, params) {
...
   const text = el.textContent?.replace(/^CTA +/, '');
   const cta = service.createCheckoutLink(context, text);
-  cta.classList.add('con-button');
-  cta.classList.toggle('button-l', large);
-  cta.classList.toggle('blue', strong);
+  if (isCta) {
+    cta.classList.add('con-button');
+    cta.classList.toggle('button-l', large);
```
[the remark this answers is not in the document]
Comment (VKniaz) [reply libs/blocks/merch/merch.js:133]: good catch

**Pull request merged**